### DISCUSSION TITLE
Headless deploy for peer innernet

### DISCUSSION THREAD

**Discussion by AlexMikhalev** [tonarino/innernet#245] [Q&A]
Hello all,
I came across your amazing project and blog post. 
I am wondering how you add humanless machines to your network.
Can you point to an example? I am thinking about using innernet to route traffic to firecracker VMs.

Regards,

Alex

**Comment by strohel** [comment]
Hi Alex, I'm not sure if I fully understand your question: innernet currently expects human interaction when adding a new peer [1], but then it can be set up (i.e. using systemd) to be completely non-interactive.

And adding a new peer could be scripted too, these are just commands in the end (if you specify all needed parameters and `--yes`, then commands should become non-interactive - unless we have an omission somewhere).

[1] creating peer from another admin peer, sending the invitation + receiving and installing the invitation on the new peer

**Comment by AlexMikhalev** [comment]
In the readme there is an example: "The name is ci-servers and the CIDR is 10.60.64.0/24, but for this example it can be anything." Let's say I have a bunch of VMs which I can run on the same server as the main node, will connecting those VMs allow communication on infrastructure layer how it is in the blog:
```
10.80.0.0/15 tonari
    10.80.1.0/24 infrastructure
    10.80.64.0/18 humans
        10.80.64.0/23 engineering
        10.80.66.0/23 non-engineering
		10.80.128.0/20 tooling
    10.81.0.0/16 installations
        10.81.0.0/24 straylight
        10.81.1.0/24 frontier
        10.81.2.0/24 mars
```
the aim is to have fully automated VMs provisioning and decommissioning - CI example fits perfectly.

**Reply by bschwind** [reply]
Hi @AlexMikhalev 

> will connecting those VMs allow communication on infrastructure layer how it is in the blog

If the peers are in CIDRs, and those CIDRs are "associated" (via `innernet add-association`), then they should be able to communicate with each other. Of course, their external endpoints also need to be accessible in some way.

As @strohel mentioned, you can pass the `--yes` flag to most commands to make them scriptable.

Hope that answers your question!
